core/tests/protractor_desktop/topicAndStoryEditor.js

```diff
@@ -197,3 +197,160 @@ describe('Topic editor functionality', function() {
+    var ids = [];
+    for (var i = 0; i < number; i++) {
+      (function(i) {
+        var info = dummyExplorationInfo.slice();
+        info[0] += i.toString();
+        workflow.createAndPublishExploration.apply(workflow, info);
+        browser.getCurrentUrl().then(function(url) {
+          var id = url.split('/')[4].replace('#', '');
+          ids.push(id);
+        });
+      })(i);
+    }
+    return ids;
+  };
+
+  var createDummySkills = function(number) {
+    var skills = [];
+    for (var i = 0; i < number; i++) {
+      var skillName = 'skillFromChapterEditor' + i.toString();
+      var material = 'material' + i.toString();
+      workflow.createSkillAndAssignTopic(skillName, material, 'Topic 0');
+      skills.push(skillName);
+    }
+    return skills;
+  };
+
+  beforeAll(function() {
+    topicsAndSkillsDashboardPage =
+      new TopicsAndSkillsDashboardPage.TopicsAndSkillsDashboardPage();
+    topicEditorPage = new TopicEditorPage.TopicEditorPage();
+    storyEditorPage = new StoryEditorPage.StoryEditorPage();
+    skillEditorPage = new SkillEditorPage.SkillEditorPage();
+    explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+    explorationEditorMainTab = explorationEditorPage.getMainTab();
+    users.createAndLoginAdminUser(
+      'creator@chapterTest.com', 'creatorChapterTest');
+    dummyExplorationIds = createDummyExplorations(3);
+    topicsAndSkillsDashboardPage.get();
+    topicsAndSkillsDashboardPage.createTopicWithTitle('Topic 0');
+    topicEditorPage.createStory('Story 0');
+    browser.getCurrentUrl().then(function(url) {
+      storyId = url.split('/')[4];
+      dummySkills = createDummySkills(2);
+    });
+  });
+
+  beforeEach(function() {
+    users.login('creator@chapterTest.com');
+    storyEditorPage.get(storyId);
+  });
+
+  it('should create a basic chapter.', function() {
+    storyEditorPage.createInitialChapter('Chapter 1');
+    storyEditorPage.setChapterExplorationId(dummyExplorationIds[0]);
+    storyEditorPage.changeNodeOutline(forms.toRichText('First outline'));
+    storyEditorPage.saveStory('First save');
+  });
+
+  it('should add one more chapter to the story', function() {
+    storyEditorPage.createNewDestinationChapter('Chapter 2');
+    storyEditorPage.navigateToChapterByIndex(1);
+    storyEditorPage.changeNodeOutline(forms.toRichText('Second outline'));
+    storyEditorPage.setChapterExplorationId(dummyExplorationIds[1]);
+    storyEditorPage.saveStory('First save');
+  });
+
+  it('should try to add one more chapter with existing exploration',
+    function() {
+      storyEditorPage.navigateToChapterByIndex(1);
+      storyEditorPage.createNewDestinationChapter('Chapter 3');
+      storyEditorPage.navigateToChapterByIndex(2);
+      storyEditorPage.setChapterExplorationId(dummyExplorationIds[1]);
+      storyEditorPage.expectExplorationIdAlreadyExistWarning();
+      allowedError = ['The given exploration already exists in the story.'];
```

Comment:
I know you don't actually add any more errors to this list, but appending/pushing the error message to the list would match the idea behind `allowedError` (as I understand it at least) better than assigning the list.

Reply:
Updated